frontend/tests/unit/test_callgraph.py

```diff
@@ -0,0 +1,91 @@
+import pytest
+import os
+from frontend_benchmark.user_callgraph import Callgraph
+from frontend_benchmark.data_structures import Branch
+
+
+@pytest.fixture
+def rootdir():
+    return os.path.join(
+        os.path.dirname(os.path.abspath(__file__)),
+        os.path.pardir,
+    )
+
+
+@pytest.fixture
+def resources(rootdir):
+    return os.path.join(rootdir, 'resources')
+
+
+def test_callgraph_from_proto_file(resources):
+    path = os.path.join(resources, 'cfg.pb')
+    cfg = Callgraph.from_proto(path)
+    assert len(cfg.functions.keys()) > 10
+
+
+def test_get_formatted_headers_onecallchain(resources):
+    test_file = os.path.join(resources, 'onecallchain.pbtext')
+    expected = ('void function_2();\n'
+                'void function_3();\n'
+                'void function_4();\n'
+                'void function_5();\n'
+                'void function_6();\n')
+    cfg = Callgraph.from_proto(test_file)
+    output = cfg.format_headers()
+    assert output == expected
+
+
+def test_load_onecallchain(resources):
+    test_file = os.path.join(resources, 'onecallchain.pbtext')
+    cfg = Callgraph.from_proto(test_file)
+    assert isinstance(cfg, Callgraph)
+    assert len(cfg.functions) == 5
+    assert cfg.entry_point == 2
+
+
+def test_load_binary_proto(resources):
+    test_file = os.path.join(resources, 'onefunction.pb')
+    cfg = Callgraph.from_proto(test_file)
+    assert isinstance(cfg, Callgraph)
+    assert len(cfg.functions) == 1
+    assert cfg.entry_point == 2
+
+
+def test_load_onefunction(resources):
+    test_file = os.path.join(resources, 'onefunction.pbtext')
+    cfg = Callgraph.from_proto(test_file)
+    assert isinstance(cfg, Callgraph)
+    assert len(cfg.functions) == 1
+    assert cfg.entry_point == 2
+
+
+def test_print_function_onefunction(resources):
+    test_file = os.path.join(resources, 'onefunction.pbtext')
+    expected = ('void function_19() {\n'
+                'label1:;\n'
+                'int x = 1;\n'
+                'int y = x*x + 3;\n'
+                'int z = y*x + 12345;\n'
+                'int w = z*z + x - y;\n'
+                '}\n')
```

Comment:
Can we add a test for the callchain too? Just need to verify that one function calls the next function as expected.